datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx

```diff
@@ -65,19 +79,45 @@ export const DescriptionEditor = ({ onComplete }: { onComplete?: () => void }) =
         refetch?.();
     };
 
+    // Function to handle all changes in Editor
+    const handleEditorChange = (editedDescription: string) => {
+        setUpdatedDescription(editedDescription);
+        if (editedDescription === description) {
+            setIsDescriptionUpdated(false);
+        } else {
+            setIsDescriptionUpdated(true);
+        }
+    };
+
+    // Updating the localStorage when the user has paused for 5 sec
+    React.useEffect(() => {
+        let delayDebounceFn: ReturnType<typeof setTimeout>;
+        const editedDescriptionsLocal = (localStorageDictionary && JSON.parse(localStorageDictionary)) || {};
+
+        if (isDescriptionUpdated) {
+            delayDebounceFn = setTimeout(() => {
+                editedDescriptionsLocal[urn] = updatedDescription;
+                localStorage.setItem('editedDescriptions', JSON.stringify(editedDescriptionsLocal));
+            }, 5000);
+        }
+        return () => clearTimeout(delayDebounceFn);
+    }, [urn, isDescriptionUpdated, updatedDescription, localStorageDictionary]);
+
     return entityData ? (
         <>
             <TabToolbar>
-                <Button type="text" onClick={onComplete}>
-                    Cancel
-                </Button>
-                <Button onClick={handleSaveDescription}>
+                <DiscardDescriptionModal
+                    buttonProps={{ type: 'text' }}
+                    isDescriptionUpdated={isDescriptionUpdated}
+                    urn={urn}
+                />
+                <Button onClick={handleSaveDescription} disabled={!isDescriptionUpdated}>
```

Comment:
Why do we only allow saving if the description is updated? The button should always be visible

Reply:
As per the given task given by Gabe, Users can only save if there will be some change in the Editor from the last save. That's why I'm disabling the button.

 - Since save does not close,  keep track of whether a change was made. If no change has been made since the last save (or since entering edit mode), have the save button be disabled: